node/network/bridge/src/lib.rs

```diff
@@ -447,19 +462,28 @@ where
 							?peer,
 							peer_set = ?peer_set,
 						);
-						network_service.disconnect_peer(peer, peer_set);
+
+						let version = match peer_set {
+							PeerSet::Validation => shared.0.lock().validation_peers.get(&peer).map(|d| d.version),
+							PeerSet::Collation => shared.0.lock().collation_peers.get(&peer).map(|d| d.version),
+						};
+
+						if let Some(version) = version {
+							network_service.disconnect_peer(peer, peer_set, version);
+						}
 					}
 					NetworkBridgeMessage::SendValidationMessage(peers, msg) => {
 						tracing::trace!(
 							target: LOG_TARGET,
 							action = "SendValidationMessages",
-							num_messages = 1,
+							num_messages = 1usize,
 						);
 
 						send_message(
 							&mut network_service,
 							peers,
 							PeerSet::Validation,
+							1, // TODO [now]: constant / SendValidationMessageV1?
```

Comment:
Yes, and probably `enum VersionedBitfieldDistributionMessage` where `NetworkBridgeEvent::focus` automatically does the transformation.

Reply:
I revisited this (everything is just a type alias of the same `Versioned` enum, which makes code a bit easier to write as you can just do `Versioned::V1` everywhere).

I would rather have encoding handled within the network bridge through functions like `fn send_validation_message_v1` which take an explicit type. Rather than doing encoding in a way that masks the type.